app/src/main/java/com/cp/tasky/auth/login/presentation/LoginViewModel.kt

```diff
@@ -46,18 +45,18 @@ class LoginViewModel @Inject constructor(
     }
 
     private fun loginUser(email: String, password: String) {
-        _networkState.value = Result.Loading
-
         viewModelScope.launch {
+            _networkResponse.send(Result.Loading)
+
             authRepository.loginUser(
                 UserCredential(
                     email = email,
                     password = password
                 )
-            ).onSuccess { responseData ->
-                _networkState.value = Result.Success(responseData)
+            ).onSuccess {
+                _networkResponse.send(Result.Success(Unit))
             }.onError { error ->
-                _networkState.value = Result.Error(error)
+                _networkResponse.send(Result.Error(error))
```

Comment:
For error messages it depends how you show them. If you show a temporary popup that disappears again (like a Toast or Snackbar), these should be done with a Channel. If you show a persistent error message (e.g. below a text field for a validation error), that's a message that keeps on showing until a certain condition is met, so it should be state.